### PR TITLE
MNT: Re-rendered with conda-smithy 2.0.0

### PR DESCRIPTION
**Pull request by goanpeca**

### CODE REVIEW

**Comment by dopplershift** [comment]
So, speaking from experience, the odd error in the test phase is due to a problem  with flake8 reading one of the files it was given to check. Last time it was non-existence due to environment variable/path problems.

**Comment by goanpeca** [comment]
@dopplershift @jakirkham any thoughts on how to fix this?

**Comment by croth1** [comment]
It seems like that the source directory is gone now when testing starts. You could try changing the tests to:

```  commands:
    - flake8 ${SP_DIR}/flake8/checker.py  # [unix]
    - flake8 %SP_DIR%\\flake8\\checker.py  # [win]
```

**Comment by goanpeca** [comment]
@jakirkham you ok with that change?

**Comment by jakirkham** [comment]
The other option is to use [`source_files`]( http://conda.pydata.org/docs/building/meta-yaml.html#source-files ) to copy over the needed parts for testing.

**Comment by goanpeca** [comment]
Nice, did not know that one. Will add the fix

**Comment by goanpeca** [comment]
@jakirkham since conda forge is only building 35 and 36 we could get rid of the `    - enum34  # [py2k or py33]` dependency? or I just leave it there?

**Comment by goanpeca** [comment]
@dopplershift this one is ready!

**Comment by jakirkham** [comment]
Concerning `enum34`, I see no harm in keeping it as people can/do use recipes from here to build offline as well. The only real reason to adjust it is if `flake8` drops support for Python 3.3, which appears to have happened in `master`. Though I haven't checked the release we are currently packaging.

**Comment by goanpeca** [comment]
@jakirkham, the this is ready :-p

File: recipe/meta.yaml [hunk 1]
```diff
@@ -18,25 +18,31 @@ build:
...
-    - pycodestyle >=2.0,<2.3
-    - pyflakes >=1.2.3,<1.4
-    - enum34  # [py2k or py33]
-    - configparser  # [py2k]
 
 test:
   imports:
     - {{ name }}
+  source_files:
+    - src/flake8
```
Comment (jakirkham) [inline recipe/meta.yaml]: Should we include the downloaded source or use the source installed in `site-packages` per @croth1's suggestion? Not picky either way as I don't expect it matters much for this case. Just figured I'd mention it. :)

Comment (goanpeca) [inline recipe/meta.yaml]: Hmmm neeh ;-)

Comment (croth1) [inline recipe/meta.yaml]: No strong opinion here. Running flake8 on one file should already be enough to make sure we got the dependencies right.

Comment (jakirkham) [inline recipe/meta.yaml]: Yep, agreed. Think you should be fine either way.

**Comment by goanpeca** [comment]
Someone will have to merge this @jakirkham?

**Comment by croth1** [comment]
thanks a lot for your work here.

**Comment by jakirkham** [comment]
Figured I'd let one of the official maintainers do the honors. 😄

**Pull request merged**